internal/impl/aws/input_kinesis_test.go

```diff
@@ -148,3 +182,103 @@ func TestRunMaxShardConfigBalancedShards(t *testing.T) {
 	}
 
 }
+
+func createKinesisClientClaim(numShards int) []awsKinesisClientClaim {
+	var clientClaims []awsKinesisClientClaim
+	for i := 0; i < numShards; i++ {
+		clientClaims = append(clientClaims, awsKinesisClientClaim{ShardID: strconv.Itoa(i), LeaseTimeout: time.Now()})
+	}
+	return clientClaims
+}
+
+func TestStealShard(t *testing.T) {
+
+	testCases := []struct {
+		testName          string
+		streams           []string
+		streamsMaxShard   []string
+		clientClaims      map[string][]awsKinesisClientClaim
+		streamNameToClaim string
+		clientId          string
+		expectedIsError   bool
+		errShardNotStolen bool
```

Comment:
Could we combine this two fields to `expectedError` of type `error`. Then in test cases it can either be `nil` or `errShardNotStolen`

Reply:
i know in my test cases that both are the same - but in the future there will be scenarios where we do expect an error but it is NOT `errShardNotStolen`

Reply:
The problem imho stems from the change that we made making `errShardNotStolen` an error, which _technically_ it is not